middleware/grpc/server.go

```diff
@@ -81,7 +91,23 @@ func (s *serverHandler) TagRPC(ctx context.Context, rti *stats.RPCTagInfo) conte
 
 	sc := s.tracer.Extract(b3.ExtractGRPC(&md))
 
-	span := s.tracer.StartSpan(name, zipkin.Kind(model.Server), zipkin.Parent(sc), zipkin.RemoteEndpoint(remoteEndpointFromContext(ctx, "")))
+	// store whitelisted headers to be propagated in spanContext
+	if s.baggage != nil {
+		sc.Baggage = s.baggage.Init()
+		sc.Baggage.IterateWhiteList(func(key string) {
+			vals := md.Get(key)
+			if len(vals) > 0 {
+				sc.Baggage.AddHeader(key, vals...)
```

Comment:
unrelated, but interesting you are supporting multiple values. in brave we overwrote the baggage (ex so you can reset a value without relying on last wins). Unless `Add` implies overwriting the key...

Reply:
no, overwriting can happen by doing explicit Delete and then AddHeader... the AddHeader is additive... Maybe add a SetHeader for overwrite? This would follow common patterns found like in gRPC metadata handling